confluence-xml/src/main/java/org/xwiki/contrib/confluence/filter/internal/input/ConfluenceInputFilterStream.java

```diff
@@ -823,6 +872,17 @@ private void readPageRevision(long pageId, String spaceKey, ConfluenceProperties
             readPageComment(pageId, spaceKey, proxyFilter, commentId, pageComments, commentIndeces);
         }
 
+        // Blog home page
+        if (this.properties.isBlogsEnabled()
```

Comment:
Shouldn't readPage not be called in the first page if this.properties.isBlogsEnabled() is false ? Or do you really want to import the blog as normal page when blog import is disabled ?